pySDC/implementations/problem_classes/DiscontinuousTestODE.py

```diff
@@ -0,0 +1,221 @@
+
+        if self.nvars != 1:
+            raise ParameterError('nvars has to be equal to 1!')
+
+        self.t_switch_exact = np.log(5)
+        self.t_switch = None
+        self.nswitches = 0
+        self.newton_itercount = 0
+        self.newton_ncalls = 0
+
+    def eval_f(self, u, t):
+        """
+        Routine to evaluate the right-hand side of the problem.
+
+        Parameters
+        ----------
+        u : dtype_u
+            Current values of the numerical solution.
+        t : float
+            Current time of the numerical solution is computed.
+
+        Returns
+        -------
+        f : dtype_f
+            The right-hand side of the problem.
+        """
+
+        t_switch = np.inf if self.t_switch is None else self.t_switch
+
+        f = self.dtype_f(self.init, val=0.0)
+        h = u[0] - 5
+        if h >= 0 or t >= t_switch:
+            f[:] = 4 / self.t_switch_exact
+        else:
+            f[:] = u
+        return f
+
+    def solve_system(self, rhs, dt, u0, t):
+        r"""
+        Simple Newton solver for :math:`(I-factor\cdot A)\vec{u}=\vec{rhs}`.
+
+        Parameters
+        ----------
+        rhs : dtype_f
+            Right-hand side for the linear system.
+        dt : float
+            Abbrev. for the local stepsize (or any other factor required).
+        u0 : dtype_u
+            Initial guess for the iterative solver.
+        t : float
+            Current time (e.g. for time-dependent BCs).
+
+        Returns
+        -------
+        me : dtype_u
+            The solution as mesh.
+        """
+
+        t_switch = np.inf if self.t_switch is None else self.t_switch
+
+        h = rhs[0] - 5
+        u = self.dtype_u(u0)
+
+        n = 0
+        res = 99
+        while n < self.newton_maxiter:
+            # form function g with g(u) = 0
+            if h >= 0 or t >= t_switch:
+                g = u - dt * (4 / self.t_switch_exact) - rhs
+            else:
+                g = u - dt * u - rhs
+
+            # if g is close to 0, then we are done
+            res = np.linalg.norm(g, np.inf)
+
+            if res < self.newton_tol:
+                break
+
+            if h >= 0 or t >= t_switch:
+                dg = 1
+            else:
+                dg = 1 - dt
+
+            # newton update
+            u -= 1.0 / dg * g
+
+            n += 1
+
+        if np.isnan(res) and self.stop_at_nan:
+            raise ProblemError('Newton got nan after %i iterations, aborting...' % n)
+        elif np.isnan(res):
+            self.logger.warning('Newton got nan after %i iterations...' % n)
+
+        if n == self.newton_maxiter:
+            self.logger.warning('Newton did not converge after %i iterations, error is %s' % (n, res))
+
+        self.newton_ncalls += 1
+        self.newton_itercount += n
+
+        me = self.dtype_u(self.init)
+        me[:] = u[:]
+
+        return me
+
+    def u_exact(self, t, u_init=None, t_init=None):
```

Comment:
Maybe you can get a logger for the problem and raise a warning when `u_init` or `t_init` are not None. I should add something like this to other problems with analytical exact solution because all of them will return the global error when using the local error hook without telling you about it.

Reply:
Will add that!